pkg/apis/fake/v1/db_types.go

```diff
@@ -0,0 +1,282 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DBSpec defines the desired state of DB
+type DBSpec struct {
+	// RootPassword is root account password for this data. Leave blank for auto-generation
+	RootPassword string `json:"rootPassword,omitempty"`
+
+	// DatabaseName of the created database at start up
+	// +kubebuilder:validation:MinLength=1
+	DatabaseName string `json:"databaseName"`
+
+	// DatabaseUser for created database
+	DatabaseUser string `json:"databaseUser"`
+
+	// DatabasePassword for the created database. Leave blank for auto-generation
+	DatabasePassword string `json:"databasePassword,omitempty"`
+
+	// Config is E2E tests params
+	Config Config `json:"config,omitempty"`
+}
+
+// OperationFlagType represents a enable/disable flag
+type OperationFlagType string
+
+func (o OperationFlagType) Enabled() bool {
+	return o == OperationFlagEnabled
+}
+
+// Values of OperationFlagType.
+const (
+	OperationFlagEnabled  OperationFlagType = "enabled"
```

Comment:
As we discussed in the webhook PR, can you please use strings with initial capitalized letters?

Reply:
done